### PR TITLE
Feature: Adding genomefiles api

### PR DESCRIPTION
**Pull request by govind-basepair**
**Task:** https://app.asana.com/0/577079795655810/1185245861049265/f

### CODE REVIEW

File: basepair/api.py [hunk 1]
```diff
@@ -1523,6 +1527,16 @@ def get_file_by_node(self, sample, node=None, analysis_tags=None,
...
 
+    def get_genomefile_by_filters(self, filters=None):
+        if not filters:
+            print('Filters required.', file=sys.stderr)
+            return
+        return self.get_obj_by_filters(url=self.get_genomefile_url(), filters=filters)
+
+    def get_obj_by_filters(self, url, filters=None):
+        res, code = self.get_request(url, user_params=filters)
+        return (res and res[0]) or res
```
Comment (saurabhbp) [inline basepair/api.py:1538]: is res guaranteed to be an array here?

File: basepair/api.py [hunk 2]
```diff
@@ -1523,6 +1527,16 @@ def get_file_by_node(self, sample, node=None, analysis_tags=None,
 
         return filepath
 
+    def get_genomefile_by_filters(self, filters=None):
+        if not filters:
+            print('Filters required.', file=sys.stderr)
+            return
+        return self.get_obj_by_filters(url=self.get_genomefile_url(), filters=filters)
+
+    def get_obj_by_filters(self, url, filters=None):
```
Comment (blasvicco) [inline basepair/api.py:1536]: @govind-basepair does look like there is no need of this extra method.
The whole think can be implemented in get_genomefile_by_filters.
BTW, why we are returning just the first element of a list?
If we need just one element from the API, then the limit parameter should be set to 1.

Comment (govind-basepair) [reply basepair/api.py]: Yeah. That makes sense. I have updated it

**Comment by govind-basepair** [comment]
@blasvicco please take a look at the changes when you get time

**Pull request merged**